### ISSUE TITLE
Random sentence is suboptimal

### ISSUE THREAD

**Issue by jiru** [Tatoeba/tatoeba2#732]
The [random sentence page](http://tatoeba.org/sentences/show/random) and clicking on the “show another“ link on the home page take a long time to load and `searchd`’s CPU usage is abnormally high. 

Each time a random sentence is loaded, a search is performed in Sphinx. For some reason, theses search take several seconds to complete (up to about 20 seconds under high load). Sphinx’s `query.log` contains such lines:

```
[Sat Jul 25 13:55:11.878 2015] 10.046 sec 10.046 sec [all/0/ext 4172841 (0,100)] [und_index]
```

Which means “100 first results retrieved from a search that matched 4172841 sentences and took 10 seconds”. There is apparently a cache mechanism that is supposed to prevent performing the search each time, but it’s obviously not working.

**Comment by trang** [comment]
I'm not sure if it's related but you reminded me that some time ago I had to set `Cache.disable` to `true` on the prod.

<pre>
/**
 * Turn off all caching application-wide.
 *
 */
        Configure::write('Cache.disable', true);
</pre>

(it was previously commented out)

This was after an update that introduced some changes in the `users` table. It caused some "Internal error" on the prod website. I figured it was a cache issue, but even after trying to empty the cache in the `tmp` folder and the cache of xcache, the error was still there. I set `Cache.disable` to true to take care of the problem later, but I forgot about it later on.

I've re-enabled it now.

**Comment by jiru** [comment]
It didn’t change anything. The problem is yet to be solved.

**Comment by trang** [comment]
Okay, I found it.

The `duration` in xcache config was set to `1`. I've changed it to `3600`.

@jiru I let you close this issue if you don't find any other problem.

**Comment by jiru** [comment]
Great! I confirm the cache is now working. dev.tatoeba.org still has the same problem however.

**Comment by jiru** [comment]
Note that it’s better, but not optimal. There is still one random sentence load out of 100 that will take seconds to load. In addition, if one requests a random sentence while the cache is empty and waits to be refilled, it needlessly restarts the slow query and overwrites the cache. Maybe performing a search each time but limited to one result is a better trade-off.

**Comment by trang** [comment]
Until we find a nice and fast way to get random sentences in any language, we should probably disable the random order if the user chooses `from=und` in the advanced search. Ordering by random when the `from` language is specified seems to be fine still.

I removed the "Several random sentences" feature (https://github.com/Tatoeba/tatoeba2/commit/b50b4c532447e2c7dfb804e85ba0898cdca9618f) thinking that the advanced search does the same job. But I realized that there's a difference: the "Several random sentences" was using the cache and therefore was much faster.
